test/framework/alertmanager.go

```diff
@@ -277,11 +281,11 @@ func (f *Framework) GetAlertmanagerMetrics(ns, n string) (textparse.Parser, erro
 }
 
 func (f *Framework) CreateSilence(ns, n string) (string, error) {
-	var createSilenceResponse amAPICreateSilResp
+	var createSilenceResponse models.PostableSilence
```

Comment:
Ah, my bad. The returned payload isn't a `models.PostableSilence` struct. We need [`PostSilencesOKBody`](https://github.com/prometheus/alertmanager/blob/4c6c03ebfe21009c546e4d1e9b92c371d67c021d/api/v2/client/silence/post_silences_responses.go#L162-L166) here.

Reply:
So we need to create a copy of the struct right or there is a way to import and use?